Command/RedisFlushallCommand.php

```diff
@@ -48,7 +48,10 @@ protected function executeRedisCommand()
      */
     private function flushAll()
     {
-        if (!($this->redisClient instanceof \IteratorAggregate)) { // BC for Predis 1.0
+        if (!($this->redisClient instanceof \IteratorAggregate) || // BC for Predis 1.0
+            // bug fix https://github.com/nrk/predis/issues/552
+            !($this->redisClient->getConnection() instanceof \Traversable)
```

Comment:
This condition reveals the [internal implementation](https://github.com/nrk/predis/blob/master/src/Client.php#L546) of [Predis](https://github.com/nrk/predis), but without this fix, SncRedisBundle will be broken in most user projects.

Reply:
I think it's fine if we consider it a compatibility adapter, as I deem it unlikely that Predis will be further maintained given its state.